src/main/java/seedu/address/logic/commands/schedule/DeleteEventCommand.java

```diff
@@ -0,0 +1,91 @@
+package seedu.address.logic.commands.schedule;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.EditUtil.EditPersonDescriptor;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.EditTypeCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.schedule.Schedule;
+
+
+/**
+ * Deletes an event of an existing person in the schedule of address book.
+ */
+public class DeleteEventCommand extends EditTypeCommand {
+
+    public static final String COMMAND_WORD = "deleteEvent";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes an event of the person identified "
+            + "by the index number used in the displayed person list.\n"
+            + "Parameters: INDICES (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1 2";
+
+    public static final String MESSAGE_DELETE_EVENT_SUCCESS = "%s's No.%d event deleted successfully";
+
+    private final Index targetIndex;
+    private final Index targetEventIndex;
+
+    /**
+     * @param targetIndex of the person in the filtered person list to edit
+     * @param targetEventIndex of the event in the schedule to delete
+     */
+    public DeleteEventCommand(Index targetIndex, Index targetEventIndex) {
+        this.targetIndex = targetIndex;
+        this.targetEventIndex = targetEventIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(targetIndex.getZeroBased());
+        Schedule scheduleToEdit = personToEdit.getSchedule();
+
+        if (targetEventIndex.getZeroBased() >= scheduleToEdit.getEvents().size()) {
+            throw new CommandException(MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
```

Comment:
```suggestion
            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
```
Try to keep consistency with the previous CommandException on line 52

Reply:
We can either change this or the one on line 52

Reply:
The same should be done in EditEventCommand

Reply:
Line 59 changed on the next commit.